tests/console/suseconnect_scc.pm

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Do the registration against SCC after installation
+# Maintainer: Yi Xu <yxu@suse.de>
+
+use strict;
+use base 'y2logsstep';
+
+use testapi;
+use registration;
+use utils 'sle_version_at_least';
+
+sub run {
+    return if get_var('HDD_SCC_REGISTERED');
+    my $version  = get_required_var('VERSION');
+    my $arch     = get_required_var('ARCH');
+    my $reg_code = get_required_var('SCC_REGCODE');
+    my $scc_url  = get_required_var('SCC_URL');
+    my @addons;
+    if (get_var('SCC_ADDONS')) {
+        @addons = split(/,/, get_required_var('SCC_ADDONS', ''));    # base,script,desktop,serverapp
+    }
+    else {
+        die "Cannot get SCC_ADDONS!";
```

Comment:
If you just die in case of not defined then just use get_required_var

Reply:
No, didn't die here. It's syntax error (array handling).